src/Form/existing-login-detection.js

```diff
@@ -0,0 +1,197 @@
+//
+//  login-detection.js
+//
+//  Copyright © 2021 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+'use strict';
+
+(function () {
+    if (!window.__ddg__) {
+        Object.defineProperty(window, '__ddg__', {
+            enumerable: false,
+            configurable: false,
+            writable: false,
+            value: {
+            }
+        })
+    }
+
+    // Set `logger = console` to print messages to the web inspector console
+    const logger = window.duckduckgoDebugMessaging
+
+    function loginAttemptDetected () {
+        try {
+            logger.log('Possible login attempt detected')
+
+            window.webkit.messageHandlers.loginFormDetected.postMessage({})
+        } catch (error) {}
+    }
+
+    function inputVisible (input) {
+        return !(input.offsetWidth === 0 && input.offsetHeight === 0) && !input.hidden && input.value !== ''
+    }
+
+    function validatePasswordField (passwords) {
+        for (var i = 0; i < passwords.length; i++) {
+            var password = passwords[i]
+            var found = inputVisible(password)
+            if (found) {
+                loginAttemptDetected()
+                return found
+            }
+        }
+    }
+
+    function checkIsLoginForm (form) {
+        logger.log('Checking for login form ' + form)
+
+        var inputs = form.getElementsByTagName('input')
+        if (!inputs) {
+            return
+        }
+
+        let username, password
+
+        for (var i = 0; i < inputs.length; i++) {
+            var input = inputs.item(i)
+
+            if (input.type === 'password' && inputVisible(input)) {
+                password = input.value
+                loginAttemptDetected()
+            }
+
+            if (input.type === 'email' || input.type === 'text') {
+                username = input.value
+            }
+
+            if (password && username) {
+                window.webkit.messageHandlers.pmHandlerStoreCredentials.postMessage({username, password})
+            }
```

Comment:
This is the new logic we use to send credentials to the native layer. Again, this is temporary and will be removed when a more robust solution is in place.

Reply:
@GioSensation @brindy Just so we don't forget about this, should it be being called through `wkSend` rather than directly?